### PR TITLE
Substitute log with tracing lib

### PR DESCRIPTION
**Pull request by i1i1**
https://jira.hyperledger.org/browse/IR-1053
Signed-off-by: i1i1 <vanyarybin1@live.ru>

### Description of the Change

Remove log crate as dependency and add tracing instead.

### Benefits

Tracing has some unique features like spans and structured logging. They can be used for searching in typed logs (for example, we can do logging in such way, that it would be easy to search and track specific request or block or transaction).

### Possible Drawbacks 

It would require some more work on tracing subscribers to export logs to some format, but in the end it will help on get good and observable application.

### CODE REVIEW

File: iroha/src/block_sync.rs [hunk 1]
```diff
@@ -64,31 +64,36 @@ impl BlockSynchronizer {
...
+                                .sorted_peers()
+                                .iter()
+                                .map(|peer| message.clone().send_to(peer)),
+                        )
+                        .await,
+                    );
+                }
             }
-        }));
+            .in_current_span(),
```
Comment (rkharisov) [inline iroha/src/block_sync.rs:89]: Calling this method allow to log something in another thread with the same span as in current thread, am I right?

File: iroha/src/lib.rs [hunk 2]
```diff
@@ -95,8 +96,8 @@ pub struct Iroha {
 impl Iroha {
     /// Default `Iroha` constructor used to build it based on the provided `Configuration`.
     pub fn new(config: &Configuration, permissions_validator: PermissionsValidatorBox) -> Self {
-        iroha_logger::init(&config.logger_configuration).expect("Failed to initialize logger.");
-        log::info!("Configuration: {:?}", config);
+        iroha_logger::init(config.logger_configuration);
+        iroha_logger::info!(?config, "Started");
```
Comment (rkharisov) [inline iroha/src/lib.rs:100]: Is it a correct syntax?

**Review by rkharisov — APPROVED** [review]

File: iroha/src/lib.rs (hunk 2 above)
Comment (i1i1) [reply iroha/src/lib.rs:100]: Yeah, it is. More info could be found here:
https://docs.rs/tracing/0.1.25/tracing/#recording-fields

File: iroha/src/block_sync.rs (hunk 1 above)
Comment (i1i1) [reply iroha/src/block_sync.rs:89]: Yes!

File: iroha/src/lib.rs (hunk 2 above)
Comment (eadventurous) [inline iroha/src/lib.rs:100]:
```suggestion
        iroha_logger::info!(?config, "Loaded configuration");
```

File: iroha/src/block_sync.rs [hunk 3]
```diff
@@ -203,13 +209,13 @@ pub mod message {
                                 .send_to(peer)
                                 .await
                         {
-                            log::warn!("Failed to request blocks: {:?}", err)
+                            warn!("Failed to request blocks: {:?}", err)
```
Comment (eadventurous) [inline iroha/src/block_sync.rs:212]: There is some inconsistency in the usage style of macros, you either use a raw `warn` or `iroha_logger::warn` in different places. Please consider using them everywhere in the same way, according to our codebase style `iroha_logger::warn` is prefered.

File: iroha_client/src/client.rs [hunk 4]
```diff
@@ -99,7 +99,7 @@ impl Client {
     ///
     /// # Errors
     /// Fails if sending transaction to peer fails or if it response with error
-    #[log]
+    #[iroha_logger::log]
```
Comment (eadventurous) [inline iroha_client/src/client.rs:102]: You can leave it as `log` you already imported at the top level.

**Review by eadventurous — APPROVED** [review]

**Pull request merged**